### PR TITLE
Compiler flags

### PR DESCRIPTION
**Pull request by DvirDukhan**
this PR adds compiler flag `-Wall -Werror` and the resulting fixing due to those flags

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +47,8 @@ endif()
 
 add_subdirectory(VecSim/spaces)
 
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall")
```
Comment (rafie) [inline src/CMakeLists.txt:51]: Don't you want to set those in a somewhat more global location? (`build/cmake/common.cmake` or something)
Same note for spaces.

Comment (DvirDukhan) [reply src/CMakeLists.txt:51]: no
as this will build the google deps with those flags, and there are warnings there.

File: src/VecSim/vec_sim_index.h [hunk 2]
```diff
@@ -9,7 +9,7 @@
  * @brief Abstract C++ class for vector index, delete and lookup
  *
  */
-class VecSimIndex : public VecsimBaseObject {
+struct VecSimIndex : public VecsimBaseObject {
```
Comment (alonre24) [inline src/VecSim/vec_sim_index.h:12]: Why this change? since everything is public?

File: src/VecSim/algorithms/hnsw/hnswlib.h [hunk 3]
```diff
@@ -60,14 +63,14 @@ class HierarchicalNSW : public VecsimBaseObject {
 
     // Index state
     size_t cur_element_count;
-    int max_id;
-    int maxlevel_;
+    tableint max_id;
+    size_t maxlevel_;
```
Comment (alonre24) [inline src/VecSim/algorithms/hnsw/hnswlib.h:67]: Why `maxlevel_` is `size_t` and not `unsigned int`? It shouldn't be larger than the number of ids, and we use `unsigned int` for that...

**Review by alonre24 — COMMENTED** [review]
LGTM

**Review by alonre24 — APPROVED** [review]
